### ISSUE TITLE
FR: max radius for E4L Spinner

### ISSUE THREAD

**Issue by hems** [EnvelopSound/EnvelopForLive#79]
Hello guys, thank you for this amazing library!

I'm playing around with E4L Spinner and i really enjoy it!

One thing feature i would love to have is a "radius" for the horizontal spin ( there is even space for it on the device : D )

Similar to what we have with the Max Elev.. So the horizontal rotation isn't super extreme... ?

Currently i'm working around that limitation by using the E4L Source Panner and my own LFO and i'm wondering if that's the best way to do it now?

Thank you again!

![image](https://user-images.githubusercontent.com/27327/142778597-bab80cf4-0a5f-4d9c-9118-cce8ae27763e.png)

**Comment by roddylindsay** [comment]
Great idea! Feel free to put up a pull request! :)

**Comment by mcslee** [comment]
Definitely a reasonable idea - but just want to clarify that the horizontal and vertical rotation in the Spinner device are not equivalent. The vertical oscillation is a sinusoidal oscillation between a minimum and maximum elevation. Thus, it has a total amplitude which can be controlled - to be more or less extreme.

The horizontal spin is a 360-degree rotation. Changing its radius will not make it more or less extreme, it will mainly just be louder or quieter, as the distance to listening position changes - but the sound will still be wrapping around the listener's head at the same rate.

Just want to make sure that's understood - still seems reasonable to add a radius control, as that would provide another interesting dimension of real-time control over the sound.

**Comment by hems** [comment]
> Just want to make sure that's understood - still seems reasonable to add a radius control, as that would provide another interesting dimension of real-time control over the sound.

sure, got it!

Also i suggest we change the min value of the range to be about 0.01 Hz instead of 1hz... I been experimenting rotation on an Octo system and i find even at the lowest speeds it still not slow enough if you're making drones / instalation sort of sounds..

Of course then again i could use a E4L Source Panner with my own LFO but then the E4L Spinner gets no use :(

In an ideal world we would give some love to E4L Spinner and have a way of:

 - adjust the radius
 - making it mucher slower 
 - perhaps have an option to sync to clock?

Is that too much to ask? Perhaps i should just stick to using the Source Panner with my own LFO instead of asking for more on the Spinner?

What are your thoughts?

**Comment by mcslee** [comment]
> Is that too much to ask? Perhaps i should just stick to using the Source Panner with my own LFO instead of asking for more on the Spinner?

Yeah this is really just duplicating functionality that is available in the E4L Mono Panner. That device has a built-in LFO, its rate goes down much lower, and has an additional speed multiplier that goes -1 to 1, which can be used to change its direction or to create really slow LFOs (e.g. 0.1Hz * .1 speed will get you 0.01 Hz).

It also has Sync mode (click the Freq button). The LFO also has Cycle vs. Sin mode if you don't want the spinner to go around all 360 degrees but would rather it just oscillate over a range.

So really I would probabaly just recommend using this device instead of the Spinner, I think it has everything you want.

<img width="285" alt="Screen Shot 2021-11-22 at 7 18 04 PM" src="https://user-images.githubusercontent.com/597850/142921949-da00d8c1-54b8-4558-b47d-4b75a59f4011.png">

**Comment by mcslee** [comment]
Note that these features are also available in the E4L Stereo Panner (and the E4L Quad Panner) if you want to work with a stereo source using L+R separation.

**Comment by hems** [comment]
> So really I would probabaly just recommend using this device instead of the Spinner, I think it has everything you want.

Doesn't have radius tough, which is kinda of the most important ask from this issue.

I'm wondering tough, is radius just regular "gain"? because if it's just regular gain i can just add utility plugin and control it on a macro for instance...


Also: if those features are available on Mono Panner, Stereo Panner and Quad Panner why do we need the E4L Spinner? Is there anything the Spinner does that the other devices don't do?

**Comment by mcslee** [comment]
> Doesn't have radius tough, which is kinda of the most important ask from this issue.

The radius knob is just on the bottom row there. The LFO in both the Spinner and the Panner is only applied to the azmiuth, you can change the radius in parallel on the Panner devices.

<img width="285" alt="142921949-da00d8c1-54b8-4558-b47d-4b75a59f4011" src="https://user-images.githubusercontent.com/597850/142984424-b0b6f863-0a51-468c-9423-796c5ba62b73.png">

> I'm wondering tough, is radius just regular "gain"? because if it's just regular gain i can just add utility plugin and control it on a macro for instance...

For practical purposes, the answer to this is mainly yes. Some devices will also introduce delay to simulate the speed of sound over distance for large radius values, but I am pretty sure the library we are using is not doing that, so effectively radius is principally a gain curve.

> Also: if those features are available on Mono Panner, Stereo Panner and Quad Panner why do we need the E4L Spinner? Is there anything the Spinner does that the other devices don't do?

The Spinner was one of our first devices ever created, designed more for super-fast warbling effects from high-rate spinning. Its features are pretty much subsumed by the other devices at this point, but since it has one of the longest histories it's been kept around for posterity and sentimentality (and also its fun super-simple controls when you just want the basic fast spinning effect).

**Comment by hems** [comment]
> > Doesn't have radius tough, which is kinda of the most important ask from this issue.
> 
> The radius knob is just on the bottom row there. The LFO in both the Spinner and the Panner is only applied to the azmiuth, you can change the radius in parallel on the Panner devices.
> 
> <img alt="142921949-da00d8c1-54b8-4558-b47d-4b75a59f4011" width="285" src="https://user-images.githubusercontent.com/597850/142984424-b0b6f863-0a51-468c-9423-796c5ba62b73.png">
> 
> > I'm wondering tough, is radius just regular "gain"? because if it's just regular gain i can just add utility plugin and control it on a macro for instance...
> 
> For practical purposes, the answer to this is mainly yes. Some devices will also introduce delay to simulate the speed of sound over distance for large radius values, but I am pretty sure the library we are using is not doing that, so effectively radius is principally a gain curve.
> 
> > Also: if those features are available on Mono Panner, Stereo Panner and Quad Panner why do we need the E4L Spinner? Is there anything the Spinner does that the other devices don't do?
> 

Thanks for your detailed answers!! 

> The Spinner was one of our first devices ever created, designed more for super-fast warbling effects from high-rate spinning. Its features are pretty much subsumed by the other devices at this point, but since it has one of the longest histories it's been kept around for posterity and sentimentality (and also its fun super-simple controls when you just want the basic fast spinning effect).

Perhaps it's time then to give some love to it by adding the "radius" parameter to it : D

**Comment by mcslee** [comment]
As you wish! :)
https://github.com/EnvelopSound/EnvelopForLive/commit/f4febee4f02a03395582ca941d2b0c40284bd45c

<img width="285" alt="Screen Shot 2021-11-24 at 8 51 02 AM" src="https://user-images.githubusercontent.com/597850/143205300-74b3af6c-8deb-4968-a956-33db5a6858bd.png">

**Comment by hems** [comment]
> 

that's so sweet of you! thanks <3
